bin/conform-templates.php

```diff
@@ -0,0 +1,299 @@
+}
+
+// composer.json
+// - checks repository description
+// - checks order of sections
+// - removes default type library
+// - updates scripts
+// - updates license
+// - updates support links
+// - updates "config"
+// - removes "minimum-stability" and "prefer-stable"
+// - checks keywords
+// @todo: check branch-alias
+
+$templateContent = json_decode(file_get_contents(__DIR__ . '/../template/composer.json'), true);
+
+$sectionOrder = [];
+foreach ($templateContent as $section => $value) {
+    $sectionOrder[] = $section;
+}
+
+$content = $composer;
+if (isset($content['type']) && $content['type'] === 'library') {
+    unset($content['type']);
+}
+$content['license'] = $templateContent['license'];
+$content['support'] = $templateContent['support'];
+if (! $hasDocs) {
+    unset($content['support']['docs']);
+}
+foreach ($content['support'] as &$supportLink) {
+    $supportLink = strtr($supportLink, $replace);
+}
+
+$content['config'] = $templateContent['config'];
+$content['scripts'] = $templateContent['scripts'];
+
+if (empty($content['keywords'])) {
+    fwrite(STDERR, 'Missing "keywords" in composer.json');
+} else {
+    $hasZf = false;
+    $hasZendframework = false;
+    foreach ($content['keywords'] as &$keyword) {
+        if ($keyword === 'zf2') {
+            $keyword = 'zf';
+            $hasZf = true;
+        } elseif ($keyword === 'zf') {
+            $hasZf = 'zf';
+        } elseif ($keyword === 'zendframework') {
+            $hasZendframework = true;
+        }
+    }
+
+    if (! $hasZendframework) {
+        array_unshift($content['keywords'], 'zendframework');
+    }
+    if (! $hasZf) {
+        array_unshift($content['keywords'], 'zf');
+    }
+}
+
+if (file_exists('.docheader')) {
+    array_unshift($content['scripts']['check'], '@license-check');
+    $content['scripts']['license-check'] = 'docheader check src/ test/';
+}
+
+unset($content['minimum-stability'], $content['prefer-stable']);
+
+$list = json_decode(
+    file_get_contents('https://docs.zendframework.com/zf-mkdoc-theme/scripts/zf-component-list.json'),
+    true
+);
+
+$description = null;
+foreach ($list as $component) {
+    if (strpos($component['url'], '/' . $repo . '/') !== false) {
+        $description = rtrim($component['description'], '.');
+        break;
+    }
+}
+
+if ($description !== null) {
+    $content['description'] = $description;
+}
+
+// sort section in composer:
+
+uksort($content, function ($a, $b) use ($sectionOrder) {
+    $ia = array_search($a, $sectionOrder);
+    $ib = array_search($b, $sectionOrder);
+
+    if ($ia === $ib) {
+        return 0;
+    }
+
+    if ($ia === false) {
+        return 1;
+    }
+
+    if ($ib === false) {
+        return -1;
+    }
+
+    if ($ia < $ib) {
+        return -1;
+    }
+
+    return 1;
+});
+
+file_put_contents(
+    'composer.json',
+    json_encode(
+            $content,
+            JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES
+    ) . PHP_EOL
+);
```

Comment:
After this is done, we need to run a `composer update --lock`; otherwise, composer will complain about the lockfile being out-of-sync with the `composer.json` file. (I've experienced this a number of times...)

Reply:
Actually not, because we are not changing any requirements, I've checked it with 1.5.2 version.

Reply:
That must be a recent fix in the 1.5 series, then. If that's the case, no change needed.